[llm] Add sentencepiece tokenizer support to llm runner

larryliu0820 opened the pull request:
Fixes #11618

Add sentencepiece tokenizer support

jackzhxng reviewed (commented):
I think @kimishpatel said he had to make some additional changes on top of this to get it to work with Gemma3

guangy10 reviewed (commented):
Will it be added automatically to the apps, e.g. benchmark apps? Particular for iOS app, do we need to add the source to the project tree?

larryliu0820 commented:
> Will it be added automatically to the apps, e.g. benchmark apps? Particular for iOS app, do we need to add the source to the project tree?

Not sure, will check CI

guangy10 commented:
> > Will it be added automatically to the apps, e.g. benchmark apps? Particular for iOS app, do we need to add the source to the project tree?
> 
> Not sure, will check CI

Gemma-3 is disabled from running on the app. To check on CI, you can add gemma-3 back here: https://github.com/pytorch/executorch/blob/7503bb36ca7344faffbd3f33fd0d8eaaeaddf66d/.github/workflows/apple-perf-private-device-experiment.yml#L60, then schedule an ondemand run on your PR

kimishpatel commented:
do validate against hf tokenizer + model's output

larryliu0820 committed:
Add sentencepiece tokenizer support to llm runner (#11645)

Summary:
X-link: https://github.com/pytorch-labs/tokenizers/pull/85

Fixes https://github.com/pytorch/executorch/issues/11618

Add sentencepiece tokenizer support

Pull Request resolved: https://github.com/pytorch/executorch/pull/11645

Reviewed By: guangy10

Differential Revision: D76789606

Pulled By: larryliu0820
